packages/json-rpc-provider-proxy/src/subscription-manager/subscription-manager.ts

```diff
@@ -39,18 +43,18 @@ export const addSubscription = ({
       if (!result) return
       active.delete(result.id)
     },
-    onAbort(): Array<JsonMessage> {
-      preActive.clear()
-      const result = [...active].map(onAbort)
-      active.clear()
-      return result
+    onDisconnect,
+    onAbort() {
+      const activeCopy = [...active]
+      onDisconnect()
+      ;[...activeCopy].forEach(onAbort)
```

Comment:
if we already copied active why do we need to spread it again here?

Reply:
we don't, that's silly, thanks!